### PR TITLE
Change Referral Pathway for TB when consumables not available: Try X-ray at level 1b and then 2, and then clinical diagnosis

### LINKED ISSUE #1090: actual_appt_footprint of HSI_Tb_Xray_level2 in conflict with mode_appt_constraints = 2
In HSI_Tb_Xray_level2, if consumables are not available the HSI defaults to a clinical diagnosis, with adds another clinical appt footprint ("Under5OPD" : 1) to the expected one. Under5OPD relies on different medical officers than the expected_appt_footprint ("DiagRadio": 1);  an issue in mode_appt_constraints = 2 therefore occurs when capabilities for the officers required for the Under5OPD have already been exhausted before the HSI is ran, since we cannot know that those officers would be required until the HSI is actually ran. This is what leads to issue #1089.

There are two possible solutions:
1. Since this seems to be quite an infrequent occurrence (~3 times a month in a 50,000 population according to analysis in #1089, although this would depend on prioritisation policy) we could ignore the exhausted capabilities issue for the particular instance of the Under5OPD appt for this HSI.
2. For the specific case of HSI_Tb_Xray_level2, when checking if capabilities are still available we can, in addition to capabilities required for the expected_footprint, check whether there are also capabilities for an Under5OPD appt, whether those would actually be required when running the event or not.
3. Modify the HSI s.t. instead of simply adding the Under5OPD footprint, an additional clinical HSI is scheduled.

@tdm32, what would you recommend?

### PR DESCRIPTION
**Pull request by tdm32**
Fixes #1090

This fix addresses issue #1090 where the actual footprint returned by the TB xray HSIs can vary in terms of officer types required. This becomes an issue under mode_appt_constraints = 2 when officer time is constrained.

The TB diagnostic triage process has changed slightly to accommodate this:
- eligible children under 5 are first scheduled HSI_Tb_Xray_level1b
- if consumables are not available, a blank footprint is returned and person is referred to HSI_Tb_Xray_level2
- if no consumables available at level 2, a blank footprint is returned and person is referred to HSI_Tb_ClinicalDiagnosis 
- HSI_Tb_ClinicalDiagnosis is a new HSI which is a short (0.5* Under5OPD) appt. This appt does not repeat all the triage process of the full HSI_Tb_ScreeningAndRefer, which includes HIV test etc. but acts as a final diagnostic step if all other tests were not available. It refers a person to start treatment if the dx_test[tb_clinical] returns positive. 

test_tb.py tests all passing
treatment coverage rates consistent with version in master

Remaining issue is to add the new treatment ID to the policy lists (TREATMENT_ID = "Tb_Test_Clinical")

@marghe-molaro please review and let me know if this fixes the issue.

### CODE REVIEW

**Comment by marghe-molaro** [comment]
Based on a local long_run_all_diseases.py run under mode_appt_constraints = 2, this PR appears to solve issue #1090 previously reported. 
The missing treatment_ID (Tb_Test_Clinical) has now been added to all policies and assigned the same priority assumed for Tb_Test_Xray.

**Review by marghe-molaro — APPROVED** [review]

**Comment by tbhallett** [comment]
@tdm32  - there's a change in here for the Malaria write-up. I presume this is accidental so am rolling it back.

File: src/tlo/methods/tb.py [hunk 1]
```diff
@@ -1933,6 +1933,74 @@ def apply(self, person_id, squeeze_factor):
...
+        # if none of the above conditions are present, no further action
+        persons_symptoms = self.sim.modules["SymptomManager"].has_what(person_id)
+        if not any(x in self.module.symptom_list for x in persons_symptoms):
+            return self.make_appt_footprint({})
+
+        # check for presence of all symptoms (clinical diagnosis)
+        if all(x in self.module.symptom_list for x in persons_symptoms):
+            test_result = self.sim.modules["HealthSystem"].dx_manager.run_dx_test(
+                dx_tests_to_run="tb_clinical", hsi_event=self
+            )
```
Comment (tbhallett) [inline src/tlo/methods/tb.py:1984]: do you want to do anything if a person has some but not all of the symptoms?

**Comment by tdm32** [comment]
> @tdm32 - there's a change in here for the Malaria write-up. I presume this is accidental so am rolling it back.

Yes that's not intentional - thanks for changing

File: src/tlo/methods/tb.py [hunk 2]
```diff
@@ -1933,6 +1933,74 @@ def apply(self, person_id, squeeze_factor):
...
+    def apply(self, person_id, squeeze_factor):
+        """ Do the screening and referring process """
+
+        df = self.sim.population.props
+        now = self.sim.date
+        person = df.loc[person_id]
+        test_result = None
+
+        # If the person is dead or already diagnosed, do nothing do not occupy any resources
+        if not person["is_alive"] or person["tb_diagnosed"]:
```
Comment (tbhallett) [inline src/tlo/methods/tb.py:1967]: The none / all logic could be accomplished more efficiently used sets, I think:

i.e. 
```python
if not any(x in self.module.symptom_list for x in persons_symptoms)
```
and 
```python
if all(x in self.module.symptom_list for x in persons_symptoms)
```

becomes:
```python
set_of_symptoms_that_indicate_tb = set(self.module.symptom_list)
if not set_of_symptoms_that_indicate_tb.intersect(persons_symptoms):
```
and 
```python
if set_of_symptoms_that_indicate_tb.issubset(persons_symptoms):
```

File: src/tlo/methods/tb.py (hunk 1 above)
Comment (tdm32) [reply src/tlo/methods/tb.py:1984]: No, I think the clinical criteria require all 4 symptoms to be present for a clinical diagnosis

Comment (tbhallett) [reply src/tlo/methods/tb.py:1984]: OK!

**Review by tbhallett — APPROVED** [review]

**Pull request merged**